Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -0,0 +1,384 @@
+          "Stable URL is a GitHub Gist"
+        else
+          ""
+        end
+
+        return status_hash(formula, "skipped", (skip_msg.blank? ? nil : [skip_msg]), args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : skipped" \
+              "#{" - #{skip_msg}" unless skip_msg.blank?}"
+          return
+        end
+      end
+
+      false
+    end
+
+    def print_latest_version(info, args:)
+      formula_s = "#{Tty.blue}#{info["formula"]}#{Tty.reset}"
+      formula_s += " (guessed)" if !info["meta"]["livecheckable"] && args.verbose?
+
+      current_s =
+        if info["version"]["newer_than_upstream"]
+          "#{Tty.red}#{info["version"]["current"]}#{Tty.reset}"
+        else
+          info["version"]["current"]
+        end
+
+      latest_s =
+        if info["version"]["outdated"]
+          "#{Tty.green}#{info["version"]["latest"]}#{Tty.reset}"
+        else
+          info["version"]["latest"]
+        end
+
+      puts "#{formula_s} : #{current_s} ==> #{latest_s}"
+    end
+
+    def checkable_urls(formula)
+      urls = []
+      urls << formula.head.url if formula.head
+      if formula.stable
+        urls << formula.stable.url
+        urls.concat(formula.stable.mirrors)
+      end
+      urls << formula.homepage if formula.homepage
+
+      urls.compact
+    end
+
+    def preprocess_url(url)
+      # Check for GitHub repos on github.com, not AWS
+      url.sub!("github.s3.amazonaws.com", "github.com") if url.include?("github")
+
+      # Use repo from GitHub or GitLab inferred from download URL
+      if url.include?("github.com") && GITHUB_SPECIAL_CASES.none? { |sc| url.include? sc }
+        if url.include? "archive"
+          url = url.sub(%r{/archive/.*}, ".git") if url.include? "github"
+        elsif url.include? "releases"
+          url = url.sub(%r{/releases/.*}, ".git")
+        elsif url.include? "downloads"
+          url = Pathname.new(url.sub(%r{/downloads(.*)}, "\\1")).dirname.to_s+".git"
+        elsif !url.end_with?(".git")
+          # Truncate the URL at the user/repo part, if possible
+          github_repo_url = url[%r{((?:[a-z]+://)?github.com/[^/]+/[^/#]+)}]
+          url = github_repo_url unless github_repo_url.blank?
+
+          url = url[0..-2] if url.end_with?("/")
+
+          url += ".git"
+        end
+      elsif url.include?("/-/archive/")
+        url = url.sub(%r{/-/archive/.*$}i, ".git")
+      end
+
+      url
+    end
+
+    def latest_version(formula, args:)
+      has_livecheckable = formula.livecheckable?
+      livecheck = formula.livecheck
+      livecheck_regex = livecheck.regex
+      livecheck_strategy = livecheck.strategy
+      livecheck_url = livecheck.url
+
+      urls = [livecheck_url] if livecheck_url.is_a?(String) && !livecheck_url.blank?
+      urls ||= checkable_urls(formula)
+
+      if args.debug?
+        puts "\nFormula:          #{formula_name(formula, args: args)}"
```

Comment:
```suggestion
        puts
        puts "Formula:          #{formula_name(formula, args: args)}"
```

Reply:
and same elsewhere. Generally outputting strings with a `\n` is a sign another `puts` or `<~EOS` may be a better fit.